packages/webpack-cli/src/plugins/dotenv-webpack-plugin.ts

```diff
@@ -0,0 +1,210 @@
+import dotenv from "dotenv";
+import { Compiler, DefinePlugin, InputFileSystem, WebpackLogger } from "webpack";
+
+interface EnvVariables {
+  [key: string]: string;
+}
+interface DotenvConfig {
+  paths: string[];
+  prefixes?: string[];
+  systemvars?: boolean;
+  allowEmptyValues?: boolean;
+  expand?: boolean;
+  ignoreStub?: boolean;
+  safe?: boolean;
+}
+
+const interpolate = (env: string, vars: EnvVariables): string => {
+  const matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || [];
+  matches.forEach((match) => {
+    env = env.replace(match, interpolate(vars[match.replace(/\$|{|}/g, "")] || "", vars));
+  });
+  return env;
+};
+
+const isMainThreadElectron = (target: string | undefined): boolean =>
+  !!target && target.startsWith("electron") && target.endsWith("main");
+
+export class Dotenv {
+  #options: DotenvConfig;
+  #inputFileSystem!: InputFileSystem;
+  #compiler!: Compiler;
+  #cache!: any;
+  #logger: WebpackLogger;
+  constructor(
+    config: DotenvConfig = {
+      paths: process.env.NODE_ENV
+        ? [
+            ".env",
+            ...(process.env.NODE_ENV === "test" ? [] : [".env.local"]),
+            `.env.[mode]`,
+            `.env.[mode].local`,
+          ]
+        : [],
+    },
```

Comment:
We should not use this here, if I pass any option in our config, all paths will be overrided, it is wrong